operation/fhir-operation-bench/src/main/java/com/ibm/fhir/operation/bench/BenchOperation.java

```diff
@@ -0,0 +1,303 @@
+        }
+        pool.shutdown();
+        try {
+            // 60 seconds is a reasonable constant here. The typical transaction timeout
+            // is longer, but the threads/size should be adjusted to provide a reasonable
+            // response time e.g. 30 seconds for a single call, or something sub-second
+            // if multiple bench operation requests are to be submitted in parallel
+            result = pool.awaitTermination(60, TimeUnit.SECONDS);
+        } catch (InterruptedException x) {
+            result = false;
+            logger.warning("interrupted waiting for bench worker pool to complete");
+        }
+
+        if (!result) {
+            // work didn't complete, so force it. Note that this requires the
+            // worker loop to listen for thread interruption.
+            pool.shutdownNow();
+        }
+        return result;
+    }
+    
+    /**
+     * The work loop performed within one thread. Terminates when the
+     * amount of work requested by size is complete, or the thread is
+     * interrupted.
+     * 
+     * @implNote the purpose of this operation is to involve only
+     * the CPU so we don't do anything which may involve the database
+     * - hence there's no transaction active. That means we can't
+     * invoke the FHIRValidator, because it may need to load resources
+     * from the database.
+     * 
+     * @param size
+     */
+    private void workerLoop(int size) {
+
+        Resource resource = readJsonResource(EXAMPLE_RESOURCE);
+
+        // now do a bunch of work on this resource to burn some CPU. The rough
+        // goal is to have one iteration take approximately 1 second on a
+        // current (2021-ish) CPU. This is arbitrary and approximate, but makes 
+        // the 'size' value a little more intuitive in terms of how much work 
+        // is requested for a particular call. Note that if size = 0, we don't
+        // start any work and so the request becomes a very lightweight 'ping'
+        for (int i=0; i<size * 4; i++) {
+
+            ResourceFingerprintVisitor visitor = new ResourceFingerprintVisitor();
+            resource.accept(visitor);
+
+            // Do a rebuild of the resource
+            resource = resource.toBuilder().build();
+
+            // Render as JSON, then parse again
+            InputOutputByteStream ioStream = new InputOutputByteStream(DATA_BUFFER_INITIAL_SIZE);
+
+            // Serialize and compress the Resource
+            try (GZIPOutputStream zipStream = new GZIPOutputStream(ioStream.outputStream())) {
+                FHIRGenerator.generator(Format.JSON, false).generate(resource, zipStream);
+                zipStream.finish();
+
+                // Now parse the generated resource again
+                try (GZIPInputStream inStream = new GZIPInputStream(ioStream.inputStream())) {
+                    FHIRParser.parser(Format.JSON).parse(new InputStreamReader(inStream, StandardCharsets.UTF_8));
```

Comment:
no need to wrap the inStream (our Parser does that automatically if you pass an InputStream)

Reply:
ok